### ISSUE TITLE
[Bugfix] Close the Logp contract for Qwen3-8B TP=2 CP=2 strict bitwise consistency

### ISSUE THREAD

**Issue by inaniloquentee** [Flink-ddd/RL-Engine#329]
## Background

The current Qwen3-8B TP=2 CP=2 full-chain validation on 8xH100 reaches the RL-Kernel paths on both sides, but strict bitwise train/rollout logprob consistency still fails.

Run artifact:

```text
/home/ellm/ljj/RL-Kernel-pr230-final/examples/vime_qwen3_8b_tp2_cp2/runs/qwen3-8b-tp2-cp2-h100-20260822-diagnostic2
```

Observed failure:

```text
train/train_rollout_logprob_abs_diff = 0.00813760794699192
train/train_current_rollout_logprob_abs_diff = 0.00813760794699192
train/train_old_rollout_logprob_abs_diff = 0.00813760794699192
train/train_current_rollout_logprob_max_abs_diff = 0.23483610153198242
train/train_current_rollout_logprob_mismatch_count = 96 / 128
```

`current-vs-rollout` and `old-vs-rollout` are identical, so this is not isolated to the old-logprob cache/data path.

## Evidence from readbacks

The RL-Kernel integration is active and has no fallback.

Training-side Megatron Logp contract:

```text
backend = pytorch-vocab-parallel-logp-ws2
logits_shape = [256, 76032]
logits_dtype = float32
tp_world_size = 2
cp_world_size = 2
real_vocab_size = 151936
padded_vocab_size = 152064
reduction = all_gather over tp_vocab, acc_dtype=fp32, order=global_vocab_shard_index
```

Rollout-side vLLM Logp contract:

```text
backend = pytorch-vocab-parallel-logp-ws2
source_logits_shape = [1, 151936]
source_logits_dtype = bfloat16
contract_real_vocab_size = 151936
contract_padded_vocab_size = 151936
```

Rollout-side sampled-token diagnostic also shows native vLLM logprob and current RL-Kernel selected-logp are not bitwise equivalent in the current boundary:

```text
native_vs_rlkernel_selected_diff.max = 0.2781919240951538  # prompt/special-token path
native_vs_rlkernel_selected_diff.max = 7.71600753068924e-07  # generated-token path
```

The attempted masked-vocab diagnostic was skipped because the actual vLLM logits width is 151936, not 152064. That means the rollout side is currently operating on unpadded full logits, while Megatron training owns TP-sharded padded logits with padding masked out.

## Required sufficient condition for this target

For Qwen3-8B TP=2 CP=2 strict bitwise consistency, Logp should be treated as one end-to-end semantic contract, not just as `logits -> selected_logp` after each framework has independently produced logits.

A sufficient implementation should satisfy all of the following for the compared train/rollout path:

1. Both training and rollout use the same RL-Kernel-owned Logp boundary. Preferred direction: `linear_logp`, i.e. output projection / lm_head + selected-token logprob in one semantic operator.
2. The compared path must not depend on framework-native logits producers with different dtype/layout/reduction behavior.
3. Token identity must be identical: sampled token ids, shifted training targets, response slicing, CP row ownership, and prompt/response masking must refer to the same token rows.
4. Vocab domain must be identical: real vocab, padded vocab, padding mask, and whether LSE is computed over padded or unpadded columns must be normalized into one contract.
5. Dtype and reduction policy must be identical: accumulation dtype, final-write dtype, TP reduction order, and transport must be explicitly owned by the RL-Kernel contract.
6. CP must remain a row/layout axis only and must not participate in vocab/LSE reduction.
7. Temperature/top-p/replay semantics must be either disabled or encoded into the same contract on both sides.

The current readbacks do **not** satisfy this sufficient condition because training uses fp32 TP-sharded padded logits, while rollout uses bf16 unpadded full logits, and rollout-side sampled logprob replacement is not bitwise equivalent at the current boundary.

## Acceptance criteria

This bug is fixed only when all of the following pass on 8xH100 CUDA with Qwen3-8B, TP=2, CP=2:

- [ ] Full-chain vime run invokes RL-Kernel Megatron integration and vLLM integration for Logp.
- [ ] No Logp fallback and no Triton path in the strict run.
- [ ] Logp readbacks show the same semantic contract on train and rollout: token identity, vocab domain, dtype/reduction policy, TP/CP ownership.
- [ ] If `linear_logp` is used, both train and rollout readbacks identify that boundary explicitly.
- [ ] If selected-logp-only is retained, an additional diagnostic proves the normalized input logits are bitwise identical before selected-logp.
- [ ] `train/train_rollout_logprob_abs_diff == 0`.
- [ ] `train/train_current_rollout_logprob_abs_diff == 0`.
- [ ] `train/train_old_rollout_logprob_abs_diff == 0`.
- [ ] `train/train_current_rollout_logprob_max_abs_diff == 0`.
- [ ] `train/train_current_rollout_logprob_mismatch_count == 0`.
- [ ] The validation artifact is stored under `examples/vime_qwen3_8b_tp2_cp2/runs/<run-id>/`.
- [ ] The run command and readbacks are sufficient for another developer to reproduce the result.

## Non-goals

- Do not treat matching native vLLM sampled logprobs as the only acceptance condition. The strict target is train/rollout RL-Kernel consistency. Native-vs-RL-Kernel diagnostics are useful for explaining drift, but the final contract should be RL-Kernel-owned.
- Do not loosen the threshold to tolerance-based validation for this target.
